pkg/controller/egress/store/egressgroup_test.go

```diff
@@ -0,0 +1,183 @@
+				}},
+			},
+		},
+		{
+			name:          "Delete event",
+			fieldSelector: fields.SelectorFromSet(fields.Set{"nodeName": "node-local"}),
+			operations: func(store storage.Interface) {
+				store.Create(eg1)
+				store.Update(eg2)
+			},
+			expectedEvents: []watch.Event{
+				{Type: watch.Bookmark, Object: &controlplane.EgressGroup{}},
+				{Type: watch.Added, Object: &controlplane.EgressGroup{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					GroupMembers: nil,
+				}},
+				{Type: watch.Deleted, Object: &controlplane.EgressGroup{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					GroupMembers: nil,
+				}},
+			},
+		},
+		{
+			name:          "Modify event",
+			fieldSelector: fields.SelectorFromSet(fields.Set{"nodeName": "node-local"}),
+			operations: func(store storage.Interface) {
+				store.Create(eg1)
+				store.Update(eg1)
+				store.Update(eg3)
+				store.Update(eg3)
+			},
+			expectedEvents: []watch.Event{
+				{Type: watch.Bookmark, Object: &controlplane.EgressGroup{}},
+				{Type: watch.Added, Object: &controlplane.EgressGroup{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					GroupMembers: nil,
+				}},
+				{Type: watch.Modified, Object: &controlplane.EgressGroupPatch{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					AddedGroupMembers: []controlplane.GroupMember{{Pod: pod1}},
+				}},
+			},
+		},
+		{
+			name:          "Node not Specified",
+			fieldSelector: fields.Everything(),
+			operations: func(store storage.Interface) {
+				store.Create(eg1)
+				store.Update(eg3)
+			},
+			expectedEvents: []watch.Event{
+				{Type: watch.Bookmark, Object: &controlplane.EgressGroup{}},
+				{Type: watch.Added, Object: &controlplane.EgressGroup{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					GroupMembers: nil,
+				}},
+				{Type: watch.Modified, Object: &controlplane.EgressGroupPatch{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					AddedGroupMembers: []controlplane.GroupMember{{Pod: pod1}},
+				}},
+			},
+		},
+	}
+	for _, tc := range tests {
+		store := NewEgressGroupStore()
+		w, err := store.Watch(context.Background(), "", labels.Everything(), tc.fieldSelector)
+		if err != nil {
+			t.Fatalf("Failed to watch object: %v", err)
+		}
+		tc.operations(store)
+		ch := w.ResultChan()
+		for _, expectedEvent := range tc.expectedEvents {
+			select {
+			case actualEvent := <-ch:
+				if !assert.Equal(t, expectedEvent, actualEvent) {
+					t.Errorf("Expected event %v, got %v", expectedEvent, actualEvent)
```

Comment:
this is not necessary when `assert.Equal` is used, it will dump the objects with a more friendly way when the test fails.

Reply:
Done.